atc/db/container_repository.go

```diff
@@ -140,12 +140,20 @@ func (repository *containerRepository) FindDestroyingContainers(workerName strin
 }
 
 func (repository *containerRepository) RemoveMissingContainers(gracePeriod time.Duration) (int, error) {
+	stalledWorkers, err := stalledWorkerNames(repository.conn)
```

Comment:
Heey,

I'm not very familiar with the code under `db`, but I think there's might a rece here given that we're performing those queries within the context of separate transactions - is this right? 

I wonder if this could/should be done in a single sql statement rather than performing a query, loading that in memory, and then using in a separate statement 🤔 

wdyt?

thx!

Reply:
yea, I wanted to implement this as one atomic operation, but the challenge is that you can't do a left-join inside of a DELETE so unfortunately it has to happen in two steps, unless there's something I'm missing from the sparsely-documented sql library we're using...

Reply:
oh, interesting! 

Being that the case, should we use the same transaction for that though? Is that a pattern that we use throught `db`? Or, perhaps, use subqueries (assuming that's permitted)?

thanks!!

Reply:
yea that's a good point! we have a few other places in our codebase, like [here](https://github.com/concourse/concourse/blob/master/atc/db/check.go#L139-L175) where we do multiple stuff within one transaction so it's certainly not a brand new pattern. thanks @cirocosta for the catch 😄

Reply:
we're using USING instead, inspired by https://github.com/concourse/concourse/blob/ba5a5932f1921fd8c2d22d0246dae2af3a6abd78/atc/db/worker_task_cache.go#L97

Reply:
Done